pkgs/development/tools/misc/pahole/default.nix

```diff
@@ -2,11 +2,12 @@
 
 stdenv.mkDerivation rec {
   pname = "pahole";
-  version = "1.17";
+  version = "1.20";
   src = fetchgit {
     url = "https://git.kernel.org/pub/scm/devel/pahole/pahole.git";
     rev = "v${version}";
-    sha256 = "13dxsmhpf9n2wqggf4gd6f12rm0vhv0q96jd50gkvaxzzvgpzzbc";
+    sha256 = "11q9dpfi4qj2v8z0nlf8c0079mlv10ljhh0d1yr0j4ds3saacd15";
+    fetchSubmodules = true;
```

Comment:
Please devendor libbpf.

Reply:
current 1.17 already vendors it, it's just not explicit here -- I don't mind doing it but it should be unrelated to this PR: just making it clear we're vendoring it is an improvement over vendoring it without a word already.

As things stand there is work from debian to devendor libbpf, upsteam merged a patch for it during 1.20 development cycle but it got reverted at the last minute. I'm confident enough 1.21 will be easier to devendor -- see https://www.spinics.net/lists/dwarves/msg00732.html
Well, doing it now would mean reverting https://git.kernel.org/pub/scm/devel/pahole/pahole.git/commit/?id=7943374ac574b94b00c1c1d30fd106bdf2230140 back again ourselves and probably wouldn't be too bad, but I still think it's unrelated

Reply:
bump. I just poked upstream about the reverted patch and a way forward with them, as no-one else (the original debian dev or someone else) seemed to bother on their ML.

I still don't want to include an extra revert patch as part of this PR, the vendoring is not new with the update and I could just remove this line and pretend this discussion never happened...
Sorry for the attitude but I dislike perfectionism in reviews; I'd do this for a new package and I could easily be talked into doing it in a separate PR (even if just waiting for next update would probably get it for "free"), but this PR is still an improvement in itself in my opinion.

Reply:
> I could just remove this line and pretend this discussion never happened...

Which is never a good idea.

> perfectionism in reviews; I'd do this for a new package and I could easily be talked into doing it in a separate PR (even if just waiting for next update would probably get it for "free"), but this PR is still an improvement in itself in my opinion.

I don't care if this is done in this PR or another but while we have the package rebuild when can clean up some obvious code smells to align the package more with current standards.

Reply:
Thanks, and sorry for the attitude.

In this case libbpf is a repeat offender (for example bcc also uses the `bcc-src-with-submodule` release tarball despite having had a way of using the system's libbpf since 0.12 (~Dec 2019)); so while I agree cleanups would be good I think it should be independent of simple updates and I'd rather work with upstream along the debian efforts first than duplicate work.
I'll work on this when time allows